examples/llm/src/models/layers/gpt_blocks.py

```diff
@@ -32,14 +35,32 @@ class GPTBlock(nn.Module):
     def __init__(self,
                  cfg: DictConfig,
                  causal_attn_cls,
+                 block_idx: int,
                  device: Optional[str] = None):
         super().__init__()
         if cfg.get('alibi', False):
             assert cfg.attn_impl == 'triton' or cfg.attn_impl == 'torch', 'Only triton kernel or torch supports alibi'
         self.ln_1 = nn.LayerNorm(cfg.d_model, device=device)
         self.causal_attn = causal_attn_cls(cfg, device)
         self.ln_2 = nn.LayerNorm(cfg.d_model, device=device)
-        self.mlp = GPTMLP(cfg, device=device)
+
+        use_moe = False
+        if cfg.get('moe', None):
+            num_experts = cfg.moe.get('num_experts')
+            if isinstance(num_experts, Sequence):
```

Comment:
Do we want to raise an error here if it is a sequence and `num_experts` doesn't match the `num_blocks`?